bingads_downloader/downloader.py

```diff
@@ -70,9 +72,22 @@ def download_ad_performance_data(api_client: BingReportClient):
             with tempfile.TemporaryDirectory() as tmp_dir:
                 tmp_filepath = Path(tmp_dir, relative_filepath)
                 tmp_filepath.parent.mkdir(exist_ok=True, parents=True)
-                submit_and_download(report_request, api_client, str(filepath))
-
-        current_date += datetime.timedelta(days=-1)
+                try:
+                    start_time = time.time()
+                    submit_and_download(report_request, api_client, str(filepath))
+                    print('Downloaded data for {date:%Y-%m-%d} in {elapsed:.1f} seconds'
+                          .format(date=current_date, elapsed=time.time() - start_time))
+                    current_date += datetime.timedelta(days=-1)
+                    remaining_attempts = config.total_attempts_for_single_file
+                except urllib.error.URLError as url_error:
```

Comment:
Does it try to download this day again if an exception has happened? I currently dont see a loop over the remaining attempts.

Reply:
Forgot to answer this comment, sorry. The new version includes a comment to make it clear, the retry logic is implicit in the fact the index decrement happens after the instruction causing the exception so in case of error the index ramain the same.